pennylane/templates/subroutines/grover.py

```diff
@@ -121,9 +124,26 @@ def expand(self):
                 wires=self.wires[-1],
                 work_wires=self.work_wires,
             )
+
             PauliZ(self.wires[-1])
 
             for wire in self.wires[:-1]:
                 Hadamard(wire)
 
         return tape
+
+    def _matrix(self, *params):
+        """Matrix representation of the Grover diffusion operator"""
+        n_wires = len(self.wires)
+
+        # state |0>
+        state_0 = [1, 0]
+
+        s1 = Hadamard.matrix @ state_0
```

Comment:
A minor suggestion here: we've noticed previously that hardcoding such expressions could lead to better performance. Maybe we could define the result of `H @ |0>` here already, instead of computing it every time.

Reply:
Gotcha!